Added support for Atlas and Mordhau

uilson opened the pull request:
Added support for Atlas and and 3 tests for it, this should close the [New Game Request: ATLAS #465](https://github.com/Austinb/GameQ/issues/465).

Added support for Mordhau and and 3 tests for it, this should close the [New Game Request: Mordhau #485](https://github.com/Austinb/GameQ/issues/485).

Both games don't support player lists, meaning they don't return the players name/score/ping.

Regards.

uilson committed:
Added Support for Brink.

uilson committed:
Added Support for Call of Duty.

uilson committed:
Added Support for Call of Duty: United Offensive.

uilson committed:
Added Support for Crysis.

uilson committed:
Added Support for Crysis 2.

uilson committed:
Added Support for Wolfenstein Enemy Territory.

uilson committed:
Added missing ".php" extention

Austinb committed:
Fix pmd issues dealing with using count() in loops. Hasn't been good practice for some time, finally got caught...

uilson committed:
Merge pull request #1 from Austinb/v3

latest fork from Austinb

Austinb committed:
Port property not needed

Austinb committed:
Port property not needed

Austinb committed:
Update Crysis.php

Austinb committed:
Port property not needed

uilson committed:
Merge pull request #2 from Austinb/v3

lattest fork from Austinb

uilson committed:
Merge branch 'v3' of https://github.com/uilson/GameQ into v3

uilson committed:
Added missing tests for protocols

Crysis 2 has no test because couldnt find any server online.

uilson committed:
Merge pull request #3 from Austinb/v3

latest fork from Austinb

uilson committed:
Added Swat4 Protocol and tests

Added Swat4 protocol (extends gamespy2) and its tests.

uilson committed:
Changed protocol of crysis to gamespy 3 again.

Added new testes and changed the protocol to gamespy3 again, game seems to be picky about the protocols.
As it works with gamespy2 in Gameq v2, it wasn't working with gamespy3 in v3 but now does.. and returns results more frequently than with the gamespy2 protocol (only able to get results 1 with gamespy2, while getting results with gamespy3 four out of five times)

uilson committed:
Added Arma Protocol and tests

Added Protocol for Arma (ArmA Armed Assault) and tests, this also expands the Gamespy2 protocol.

uilson committed:
Re-done tests for Coduo

Re-done the tests for coduo (extends quake3) and added 2 more tests, dont see any issue with the players names.

uilson committed:
Re-done tests, with the Crysis protocol instead of gamespy3

updated tests

Austinb committed:
Port not required

Austinb committed:
Renamed due to broken pull from earlier in the year.

Austinb committed:
Port not required

Austinb committed:
Fixed in another commit

Austinb committed:
Fixed in another commit

uilson committed:
Update Crysis2.php

uilson committed:
Merge pull request #4 from Austinb/v3

Latest fork from Austinb

uilson committed:
Added Protocol for Atlas and Tests

Added the Atlas protocol (extends Source), and 3 tests for it.

Two with the default query ports (stock ones from game, using the $port_diff, that are the tests 1 and 2) and one with a query port different from the stock game port difference (Test 2).

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/e35614bb-5f60-4b6f-9303-d50acc1208ab) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/e35614bb-5f60-4b6f-9303-d50acc1208ab) completed: **1 updated code elements**

uilson committed:
Removed extra white spaces from end of line.

white spaces at the end of 2 lines caused the ci-travis check to fail.
Removed that extra spaces.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/5a0e7721-4241-42c1-a035-3d45b9d27b6f) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/5a0e7721-4241-42c1-a035-3d45b9d27b6f) completed: **1 updated code elements**

uilson commented:
Travis-CI gives an error when it tries to validate the test nr 2, because its the one I mentioned in the pull, that has a different port from the $port_diff logic. so it tries to use the port_diff logic but the port is a custom one. 

IP: 24.112.101.83
Host port: 6344
Query port: 6342

 ̶I̶̶̶f̶̶̶ ̶̶̶y̶̶̶o̶̶̶u̶̶̶ ̶̶̶w̶̶̶a̶̶̶n̶̶̶t̶̶̶,̶̶̶ ̶̶̶I̶̶̶ ̶̶̶c̶̶̶a̶̶̶n̶̶̶ ̶̶̶d̶̶̶e̶̶̶l̶̶̶e̶̶̶t̶̶̶e̶̶̶ ̶̶̶t̶̶̶h̶̶̶i̶̶̶s̶̶̶ ̶̶̶t̶̶̶e̶̶̶s̶̶̶t̶̶̶,̶̶̶ ̶̶̶a̶̶̶n̶̶̶d̶̶̶ ̶̶̶t̶̶̶h̶̶̶e̶̶̶ ̶̶̶p̶̶̶r̶̶̶o̶̶̶b̶̶̶l̶̶̶e̶̶̶m̶̶̶ ̶̶̶i̶̶̶t̶̶̶s̶̶̶ ̶̶̶s̶̶̶o̶̶̶l̶̶̶v̶̶̶e̶̶̶d̶̶̶.̶̶̶
_**Removed the test, and added the Mordhau protocol and tests.**_
---
Am I doing something wrong? 
Because Travis-CI/PHPUnit ignores the query port specified in the tests, and tries the port used in the "port_client" as the query port.
I use this to test, when the query port is diferent:
`php ./generate_provider -p game -s <serverip:port_client> -o '{"query_port":<query_port>}'`

should I use just:
`php ./generate_provider -p game -s <serverip:query_port>`

Or is there a way you can change the tests, so that when the _gq_port_query_ is present and is different from the _gq_port_client_, it uses it instead to validate the tests?
________
PS: also don't know why it shows all the previous commits.. as only 14 files (related to Atlas and Mordhau) where added, as it can be seen in the "Files Changed" option.. (maybe a bug in github)

uilson committed:
Added support for Mordhau

added protocol and 3 tests for Mordhau. this game does not return player lists, so no players names/score/ping will be retrieved.

removed test that braked the logic of Atlas $port_differ

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/118dfdc8-386f-44da-93b3-f85a46f58cfd) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/118dfdc8-386f-44da-93b3-f85a46f58cfd) completed: **2 updated code elements**

uilson committed:
Tests for Mordhau

New tests with the query port in the " -s "argument (next to the IP:) instead of the " -o '{"query_port":<PORT>}' " argument.
This should prevent PHPUnit from throwing errors related with different port when testing the files.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/d73437b7-63f9-4462-bba6-ea488af7b96f) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/d73437b7-63f9-4462-bba6-ea488af7b96f) completed: **2 updated code elements**
